src/site/markdown/server/manage-webadmin.md

```diff
@@ -2234,6 +2235,45 @@ Warning: During the re-indexing, the result of search operations might be altere
 
 Warning: Canceling this task should be considered unsafe as it will leave the currently reIndexed mailbox as partially indexed.
 
+### ReIndexing a single mail by messageId
+
+```
+curl -XPOST http://ip:port/index/message/{messageId}?task=reIndex
+```
+
+Will schedule a task for reIndexing a single email in all the mailboxes containing it.
+
+Note that 'messageId' path parameter needs to be a (implementation dependent) valid messageId.
+
+The response to that request will be the scheduled `taskId` :
+
+```
+{"taskId":"5641376-02ed-47bd-bcc7-76ff6262d92a"}
+```
+
+Positionned headers:
+
+ - Location header indicates the location of the resource associated with the scheduled task. Example:
+
+```
+Location: /tasks/3294a976-ce63-491e-bd52-1b6f465ed7a2
+```
+
+Response codes:
+
+ - 201: Success. Corresponding task id is returned.
+ - 400: Error in the request. Details can be found in the reported error.
+
+The scheduled task will have the following type `MessageIdReIndexingTask` and the following `additionalInformation`:
+
+```
+{
+  "messageId":"18"
```

Comment:
I prefer cassandra UUID than a number MessageId, atleast it shows the slightly difference with `uid`

Reply:
Well that depend on mailbox implementation...